cpp/src/parquet/column_writer.cc

```diff
@@ -1471,6 +1504,7 @@ Status TypedColumnWriterImpl<DType>::WriteArrowDictionary(
                       AddIfNotNull(rep_levels, offset));
     std::shared_ptr<Array> writeable_indices =
         indices->Slice(value_offset, batch_num_spaced_values);
+    update_stats(/*num_chunk_levels=*/batch_size, writeable_indices);
```

Comment:
Would `page_statistics_` be a nullptr here?

Reply:
Yes, please check: https://github.com/apache/arrow/blob/master/cpp/src/parquet/column_writer.cc#L1069

Reply:
Well, I think here we should:
`if (page_statistics) update_stats(/*num_chunk_levels=*/batch_size, writeable_indices);`

Reply:
Good catch!